src/main/java/com/box/sdk/BackoffCounter.java

```diff
@@ -22,11 +22,20 @@ public int getAttemptsRemaining() {
         return this.attemptsRemaining;
     }
 
+    /**
+     * Waits for some random amount of time.
```

Comment:
Not completely random. Maybe specify that waits for amount of time calculated by exponential backoff?